.github/workflows/build_test.yml

```diff
@@ -0,0 +1,62 @@
+name: Build/Test for PR and collaborator push
+
+on:
+  # allows us to run workflows manually
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '.github/workflows/build_test_publish.yml'
+      - '.github/workflows/changelog_test.yml'
+      - 'docker/**'
+      - 'doc/**'
+  push:
+    branches:
+      - main
```

Comment:
If we trigger on push to all branches, authors can see this be tested when they push to their own branch on their own fork.